### ISSUE TITLE
invalid unconfirmed transactions by trying open channel

### ISSUE THREAD

**Issue by smamamo** [ElementsProject/lightning#6573]
any helüp with invalid unconfirmed-transactions
hash id - b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686
How can I fix, return or cancel this invalid transaction?

cln-rtl-screenshot
<img width="1974" alt="Bildschirmfoto 2023-08-16 um 08 10 14" src="https://github.com/ElementsProject/lightning/assets/16743383/5941ed90-ce39-42d9-b48a-894a51f86b51">

<img width="1164" alt="Bildschirmfoto 2023-08-16 um 08 10 31" src="https://github.com/ElementsProject/lightning/assets/16743383/af686b5a-2d4d-467b-b981-1f4a0938751c">

**Comment by rustyrussell** [comment]
Sorry, what's wrong with the transaction? How did you create it?

**Comment by smamamo** [comment]
Hi 

Thanks for you fast response.

I try to open a channel with 1ml.com <http://1ml.com/>  but  maybe I use  to low federate. 
So never went true.

**Comment by smamamo** [comment]
Here the invalid tx in explorer

https://www.blockchain.com/explorer/transactions/btc/b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686￼

**Comment by rustyrussell** [comment]
I don't know why it says it's invalid?  But it hasn't reached other explorers that I can tell, nor my node.

CLN would have sent it to your bitcoind, which would not have relayed it if it didn't think it was valid.  Can you run `bitcoin-cli getrawtransaction b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686` ?

**Comment by smamamo** [comment]
This is the result, but I don’t know what is that means:
There was never a transaction Payment?


***@***.***:~ ₿ bitcoin-cli getrawtransaction b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686 
error code: -5
error message:
No such mempool or blockchain transaction. Use gettransaction for wallet transactions.


> Am 16.08.2023 um 12:23 schrieb Rusty Russell ***@***.***>:
> 
> bitcoin-cli getrawtransaction b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686

**Comment by ShahanaFarooqui** [comment]
@smamamo 
1- Please share your environment details, like your core lightning, c-lightning-REST and RTL versions, OS, browser etc.
2- As per your comments, it seems that the channel opening transaction was not broadcasted due to very low fee; but the utxo has been locked by the core lightning.
3- You can check the `reserved_to_block` value (If reserved is true) from `lightning-cli listfunds` to get the block height where reservation will expire and the utxo will be available again.

**Comment by smamamo** [comment]
1- Please share your environment details, like your core lightning, c-lightning-REST and RTL versions, OS, browser etc.

I run on a Mac-mini 6.1 (i7) Linux and installed RaspiBlitz v1.9.0 and Core Lightning v23.05.2 / Ride the Lightning v0.13.6
 


2- As per your comments, it seems that the channel opening transaction was not broadcasted due to very low fee; but the utxo has been locked by the core lightning.


3- You can check the reserved_to_block value (If reserved is true) from lightning-cli listfunds to get the block height where reservation will expire and the utxo will be available again.

How should I proceed for this? Teminal - comaand? 
"reserved_to_block value“ ?? 797731 ??

***@***.***:~ ₿ lightning-cli listfunds 
{
   "outputs": [
      {
         "txid": "b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686",
         "output": 1,
         "amount_msat": 4750739000,
         "scriptpubkey": "0014398b662a574919a55160a8bdbac70bce3c882bf5",
         "address": "bc1q8x9kv2jhfyv625tq4z7m43ctec7gs2l4l0r92s",
         "status": "unconfirmed",
         "reserved": false
      },
      {
         "txid": "5163fa3ebbc216a1e01dffe069d6cc3ccbc2cd6973a9264fdd9d2bf3a8dc209f",
         "output": 0,
         "amount_msat": 149489000,
         "scriptpubkey": "001470a00aa59d47fc557a6b1e490448e4c390cae883",
         "address": "bc1qwzsq4fvagl7927ntreysgj8ycwgv46yrx9hljc",
         "status": "confirmed",
         "blockheight": 797731,
         "reserved": false
      }
   ],
   "channels": [
      {
         "peer_id": "0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266",
         "connected": false,
         "state": "AWAITING_UNILATERAL",
         "channel_id": "8646684ea9bdf663435af12f6c808ea0b6feec3450aecbe172b52eabe2f332b8",
         "our_amount_msat": 248337000,
         "amount_msat": 248337000,
         "funding_txid": "b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686",
         "funding_output": 0
      }
   ]

**Comment by ShahanaFarooqui** [comment]
Hi @smamamo thanks for sharing the above. Need some more info to be able to build a clearer picture. Please share the below two items:

1. run `lightning-cli listpeerchannels` and share the output here
2. Go to the Channels page on RTL and share a screenshot of the `Pending/Inactive` tab

**Comment by saubyk** [comment]
Hi @smamamo I looked at this issue and here's my analysis based on the information available so far:

1. From the `listfunds` outputs it's clear that tx id: b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686 was used to open the channel with an amount of 248337 sats, and a change output of 4750739 sats was generated (which is appearing as unconfirmed in RTL)
2. Also from the `listfunds` output we can see that the channel is actually closing now since it is in AWAITING_UNILATERAL state
3. I was able to track the input that was used to create the above transaction, which is for the amount 4999577 sats from this link: https://www.blockchain.com/explorer/transactions/btc/b832f3e2ab2eb572e1cbae5034ecfeb6a08e806c2ff15a4363f6bda94e684686
4. Now the above input (used to fund the channel open) has already been spent on May 11th in this transaction: https://mempool.space/tx/a00e4157c5e590972455844dbe3c4e0e1aeab971a6dfe8e57f0bd53a22f9e0cd

So in conclusion, there is nothing to recover in this scenario as you have already spent the funds. You can now safely ignore this unconfirmed transaction as it will never get confirmed.

**Comment by rustyrussell** [comment]
@saubyk Thanks for the analysis!  I agree; I had missed the double-spend here.

OK, in this case there is a developer-only command which can force the forgetting of the channel (dev-forget-channel) but it'd only in developer builds.  It's simply cosmetic, and you can ignore it for now.

**Comment by smamamo** [comment]
@saubyk Thanks for the analysis! 
@rustyrussell  Thanks for you help and finally report.
